src/net.cpp

```diff
@@ -449,102 +458,108 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
     assert(!addr_bind.IsValid());
     std::unique_ptr<i2p::sam::Session> i2p_transient_session;
 
-    if (addrConnect.IsValid()) {
-        const bool use_proxy{GetProxy(addrConnect.GetNetwork(), proxy)};
-        bool proxyConnectionFailed = false;
+    for (auto& resolvedAddr: resolvedAddresses) {
```

Comment:
Here we iterate over all resolved addresses and I think this is the right thing to do as usually there are just a few, maybe IPv4 and IPv6 and some fallback addresses for redundancy.

What is the worst thing that can happen? A broken/malicious hostname could resolve to 100s addresses all of which timeout slowly, causing some delay here. I think that is ok, given that `pzsDest` never comes unchecked from random/evil peers over the P2P network and is always provided manually by the operator of the node, right?

Reply:
> What is the worst thing that can happen? A broken/malicious hostname could resolve to 100s addresses all of which timeout slowly, causing some delay here.

We are limiting the number of resolutions to 256

> I think that is ok, given that pzsDest never comes unchecked from random/evil peers over the P2P network and is always provided manually by the operator of the node, right?

That's my intuition too, yes. This is the call tree for `ConnectNode`:

```
RPC::AddConnection -->      AddConnection               --|
ThreadOpenConnections -->   ProcessAddrFetch            --|  
Connman::Start -->          ThreadOpenConnections (x2)  ------->  OpenNetworkConnection --> ConnectNode 
CConnman::Start -->         ThreadOpenAddedConnections  --|
                            RPC::AddNode ("onetry")     __|
```

Out of all these, the only method that is not purely manual is `ThreadOpenConnections`, but that only calls `OpenNetworkConnection` with `pszDest` set if connect parameters have been passed to startup, which are also manual.

Reply:
> I think that is ok, given that pzsDest never comes unchecked from random/evil peers over the P2P network and is always provided manually by the operator of the node, right?

With the slight exception that DNS seeds can, in certain circumstances, be added as `AddAddrFetch` peers and then queried via `pszDest`. These are not provided by the operator manually but hardcoded.

Reply:
I'm a bit curious about this. I think I'll try to dig into how bad could it be if a malicious DNS seed provides us with 256+ addresses that try to delay the TPC handshake as much as possible